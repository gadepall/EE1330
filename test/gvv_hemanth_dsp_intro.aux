\relax 
\providecommand\hyper@newdestlabel[2]{}
\providecommand\HyperFirstAtBeginDocument{\AtBeginDocument}
\HyperFirstAtBeginDocument{\ifx\hyper@anchor\@undefined
\global\let\oldcontentsline\contentsline
\gdef\contentsline#1#2#3#4{\oldcontentsline{#1}{#2}{#3}}
\global\let\oldnewlabel\newlabel
\gdef\newlabel#1#2{\newlabelxx{#1}#2}
\gdef\newlabelxx#1#2#3#4#5#6{\oldnewlabel{#1}{{#2}{#3}}}
\AtEndDocument{\ifx\hyper@anchor\@undefined
\let\contentsline\oldcontentsline
\let\newlabel\oldnewlabel
\fi}
\fi}
\global\let\hyper@last\relax 
\gdef\HyperFirstAtBeginDocument#1{#1}
\providecommand*\HyPL@Entry[1]{}
\bibstyle{IEEEtran}
\HyPL@Entry{0<</S/D>>}
\@writefile{toc}{\contentsline {section}{\numberline {1}Software Installation}{1}{section.1}}
\@writefile{toc}{\contentsline {section}{\numberline {2}Digital Filter}{1}{section.2}}
\newlabel{prob:input}{{1}{1}{}{problem.1}{}}
\newlabel{prob:spectrogram}{{2}{1}{}{problem.2}{}}
\newlabel{prob:output}{{3}{1}{}{problem.3}{}}
\@writefile{lol}{\contentsline {lstlisting}{Cancel\textunderscore noise.py}{1}{lstlisting.-3}}
\newlabel{eq:iir_filter}{{8.1}{2}{}{equation.8.1}{}}
\@writefile{toc}{\contentsline {section}{\numberline {3}$Z$-transform}{2}{section.3}}
\newlabel{eq:z_trans}{{9.1}{2}{}{equation.9.1}{}}
\newlabel{eq:z_trans_shift}{{9.3}{2}{}{equation.9.3}{}}
\newlabel{eq:unit_step}{{11.2}{2}{}{equation.11.2}{}}
\newlabel{eq:anun}{{12.1}{2}{}{equation.12.1}{}}
\newlabel{prob:H(z)}{{13}{2}{}{problem.13}{}}
\newlabel{prob:iir}{{15}{2}{}{problem.15}{}}
\newlabel{eq:ztransab}{{15.1}{2}{}{equation.15.1}{}}
\@writefile{toc}{\contentsline {section}{\numberline {4}Impulse Response}{2}{section.4}}
\newlabel{prob:impulse_resp}{{16}{2}{}{problem.16}{}}
\newlabel{eq:impulse_resp}{{16.1}{2}{}{equation.16.1}{}}
\newlabel{eq:iir_filter_h}{{19.1}{3}{}{equation.19.1}{}}
\newlabel{eq:convolution}{{20.1}{3}{}{equation.20.1}{}}
\@writefile{toc}{\contentsline {section}{\numberline {5}DFT and FFT}{3}{section.5}}
