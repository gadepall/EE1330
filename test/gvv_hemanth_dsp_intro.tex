\documentclass[journal,12pt,twocolumn]{IEEEtran}
%
\usepackage{setspace}
\usepackage{gensymb}
%\usepackage{graphicx}
\usepackage{xcolor}
\usepackage{caption}
%\usepackage{subcaption}
\usepackage{tabularx}
\usepackage{booktabs}
\usepackage{url}
%\doublespacing
\singlespacing

%\usepackage{graphicx}
%\usepackage{amssymb}
%\usepackage{relsize}
\usepackage[cmex10]{amsmath}
\usepackage{mathtools}

%\usepackage{amsthm}
%\interdisplaylinepenalty=2500
%\savesymbol{iint}
%\usepackage{txfonts}
%\restoresymbol{TXF}{iint}
%\usepackage{wasysym}
%\usepackage{tikz}
%\usetikzlibrary{matrix,shapes,arrows,positioning,chains}
%\usetikzlibrary{intersections}
\usepackage{textcomp}
\usepackage{amsthm}
\usepackage{mathrsfs}
\usepackage{txfonts}
\usepackage{stfloats}
\usepackage{cite}
\usepackage{cases}
\usepackage{subfig}
%\usepackage{xtab}
\usepackage{longtable}
\usepackage{multirow}
%\usepackage{algorithm}
%\usepackage{algpseudocode}
\usepackage{enumitem}
\usepackage{mathtools}
\usepackage{iithtlc}
\usepackage[breaklinks]{hyperref}
%\usepackage[framemethod=tikz]{mdframed}
\usepackage{listings}
\usepackage{listings}
    \usepackage[latin1]{inputenc}                                 %%
    \usepackage{color}                                            %%
    \usepackage{array}                                            %%
    \usepackage{longtable}                                        %%
    \usepackage{calc}                                             %%
    \usepackage{multirow}                                         %%
    \usepackage{hhline}                                           %%
    \usepackage{ifthen}                                           %%
  %optionally (for landscape tables embedded in another document): %%
    \usepackage{lscape}     
\usepackage{url}
\def\UrlBreaks{\do\/\do-}



%\usepackage{stmaryrd}


%\usepackage{wasysym}
%\newcounter{MYtempeqncnt}
\DeclareMathOperator*{\Res}{Res}
%\renewcommand{\baselinestretch}{2}
\renewcommand\thesection{\arabic{section}}
\renewcommand\thesubsection{\thesection.\arabic{subsection}}
\renewcommand\thesubsubsection{\thesubsection.\arabic{subsubsection}}

\renewcommand\thesectiondis{\arabic{section}}
\renewcommand\thesubsectiondis{\thesectiondis.\arabic{subsection}}
\renewcommand\thesubsubsectiondis{\thesubsectiondis.\arabic{subsubsection}}

% correct bad hyphenation here
\hyphenation{op-tical net-works semi-conduc-tor}

\lstset{
%language=Python,
frame=single, 
breaklines=true,
columns=fullflexible
}

%\lstset{
	%%basicstyle=\small\ttfamily\bfseries,
	%%numberstyle=\small\ttfamily,
	%language=Octave,
	%backgroundcolor=\color{white},
	%%frame=single,
	%%keywordstyle=\bfseries,
	%%breaklines=true,
	%%showstringspaces=false,
	%%xleftmargin=-10mm,
	%%aboveskip=-1mm,
	%%belowskip=0mm
%}

%\surroundwithmdframed[width=\columnwidth]{lstlisting}
\def\inputGnumericTable{}                                 %%
 
%\usepackage{hyperref}
\begin{document}
%

\theoremstyle{definition}
\newtheorem{theorem}{Theorem}[section]
\newtheorem{problem}{Problem}
\newtheorem{proposition}{Proposition}[section]
\newtheorem{lemma}{Lemma}[section]
\newtheorem{corollary}[theorem]{Corollary}
\newtheorem{example}{Example}[section]
\newtheorem{definition}{Definition}[section]
%\newtheorem{algorithm}{Algorithm}[section]
%\newtheorem{cor}{Corollary}
\newcommand{\BEQA}{\begin{eqnarray}}
\newcommand{\EEQA}{\end{eqnarray}}
\newcommand{\define}{\stackrel{\triangle}{=}}

\bibliographystyle{IEEEtran}
%\bibliographystyle{ieeetr}

\providecommand{\nCr}[2]{\,^{#1}C_{#2}} % nCr
\providecommand{\nPr}[2]{\,^{#1}P_{#2}} % nPr
\providecommand{\mbf}{\mathbf}
\providecommand{\pr}[1]{\ensuremath{\Pr\left(#1\right)}}
\providecommand{\qfunc}[1]{\ensuremath{Q\left(#1\right)}}
\providecommand{\sbrak}[1]{\ensuremath{{}\left[#1\right]}}
\providecommand{\lsbrak}[1]{\ensuremath{{}\left[#1\right.}}
\providecommand{\rsbrak}[1]{\ensuremath{{}\left.#1\right]}}
\providecommand{\brak}[1]{\ensuremath{\left(#1\right)}}
\providecommand{\lbrak}[1]{\ensuremath{\left(#1\right.}}
\providecommand{\rbrak}[1]{\ensuremath{\left.#1\right)}}
\providecommand{\cbrak}[1]{\ensuremath{\left\{#1\right\}}}
\providecommand{\lcbrak}[1]{\ensuremath{\left\{#1\right.}}
\providecommand{\rcbrak}[1]{\ensuremath{\left.#1\right\}}}
\theoremstyle{remark}
\newtheorem{rem}{Remark}
\newcommand{\sgn}{\mathop{\mathrm{sgn}}}
\providecommand{\abs}[1]{\left\vert#1\right\vert}
\providecommand{\res}[1]{\Res\displaylimits_{#1}} 
\providecommand{\norm}[1]{\lVert#1\rVert}
\providecommand{\mtx}[1]{\mathbf{#1}}
\providecommand{\mean}[1]{E\left[ #1 \right]}
\providecommand{\fourier}{\overset{\mathcal{F}}{ \rightleftharpoons}}
\providecommand{\ztrans}{\overset{\mathcal{Z}}{ \rightleftharpoons}}
%\providecommand{\hilbert}{\overset{\mathcal{H}}{ \rightleftharpoons}}
\providecommand{\system}{\overset{\mathcal{H}}{ \longleftrightarrow}}
	%\newcommand{\solution}[2]{\textbf{Solution:}{#1}}
\newcommand{\solution}{\noindent \textbf{Solution: }}
\providecommand{\dec}[2]{\ensuremath{\overset{#1}{\underset{#2}{\gtrless}}}}
%\numberwithin{equation}{subsection}
\numberwithin{equation}{problem}
%\numberwithin{problem}{subsection}
%\numberwithin{definition}{subsection}
\makeatletter
\@addtoreset{figure}{problem}
\makeatother

\let\StandardTheFigure\thefigure
%\renewcommand{\thefigure}{\theproblem.\arabic{figure}}
\renewcommand{\thefigure}{\theproblem}


%\numberwithin{figure}{subsection}

%\numberwithin{equation}{subsection}
%\numberwithin{equation}{section}
%%\numberwithin{equation}{problem}
%%\numberwithin{problem}{subsection}
%\numberwithin{problem}{section}
%%\numberwithin{definition}{subsection}
%\makeatletter
%\@addtoreset{figure}{problem}
%\makeatother
%\makeatletter
%\@addtoreset{table}{problem}
%\makeatother

%\let\StandardTheFigure\thefigure
%\let\StandardTheTable\thetable
%%\renewcommand{\thefigure}{\theproblem.\arabic{figure}}
%\renewcommand{\thefigure}{\theproblem}
%\renewcommand{\thetable}{\theproblem}
%%\numberwithin{figure}{section}

%%\numberwithin{figure}{subsection}



\def\putbox#1#2#3{\makebox[0in][l]{\makebox[#1][l]{}\raisebox{\baselineskip}[0in][0in]{\raisebox{#2}[0in][0in]{#3}}}}
     \def\rightbox#1{\makebox[0in][r]{#1}}
     \def\centbox#1{\makebox[0in]{#1}}
     \def\topbox#1{\raisebox{-\baselineskip}[0in][0in]{#1}}
     \def\midbox#1{\raisebox{-0.5\baselineskip}[0in][0in]{#1}}

\vspace{3cm}

\title{
	\logo{
Digital Signal Processing
	}
}
%\title{
%	\logo{Matrix Analysis through Octave}{\begin{center}\includegraphics[scale=.24]{tlc}\end{center}}{}{HAMDSP}
%}


% paper title
% can use linebreaks \\ within to get better formatting as desired
%\title{Matrix Analysis through Octave}
%
%
% author names and IEEE memberships
% note positions of commas and nonbreaking spaces ( ~ ) LaTeX will not break
% a structure at a ~ so this keeps an author's name from being broken across
% two lines.
% use \thanks{} to gain access to the first footnote area
% a separate \thanks must be used for each paragraph as LaTeX2e's \thanks
% was not built to handle multiple paragraphs
%

\author{ Hemanth Kumar Desineedi and G V V Sharma$^{*}$ %<-this  stops a space
\thanks{*The authors are with the Department
of Electrical Engineering, Indian Institute of Technology, Hyderabad
502285 India e-mail:  gadepall@iith.ac.in.}% <-this % stops a space
%\thanks{J. Doe and J. Doe are with Anonymous University.}% <-this % stops a space
%\thanks{Manuscript received April 19, 2005; revised January 11, 2007.}}
}
% note the % following the last \IEEEmembership and also \thanks - 
% these prevent an unwanted space from occurring between the last author name
% and the end of the author line. i.e., if you had this:
% 
% \author{....lastname \thanks{...} \thanks{...} }
%                     ^------------^------------^----Do not want these spaces!
%
% a space would be appended to the last name and could cause every name on that
% line to be shifted left slightly. This is one of those "LaTeX things". For
% instance, "\textbf{A} \textbf{B}" will typeset as "A B" not "AB". To get
% "AB" then you have to do: "\textbf{A}\textbf{B}"
% \thanks is no different in this regard, so shield the last } of each \thanks
% that ends a line with a % and do not let a space in before the next \thanks.
% Spaces after \IEEEmembership other than the last one are OK (and needed) as
% you are supposed to have spaces between the names. For what it is worth,
% this is a minor point as most people would not even notice if the said evil
% space somehow managed to creep in.



% The paper headers
%\markboth{Journal of \LaTeX\ Class Files,~Vol.~6, No.~1, January~2007}%
%{Shell \MakeLowercase{\textit{et al.}}: Bare Demo of IEEEtran.cls for Journals}
% The only time the second header will appear is for the odd numbered pages
% after the title page when using the twoside option.
% 
% *** Note that you probably will NOT want to include the author's ***
% *** name in the headers of peer review papers.                   ***
% You can use \ifCLASSOPTIONpeerreview for conditional compilation here if
% you desire.




% If you want to put a publisher's ID mark on the page you can do it like
% this:
%\IEEEpubid{0000--0000/00\$00.00~\copyright~2007 IEEE}
% Remember, if you use this you must call \IEEEpubidadjcol in the second
% column for its text to clear the IEEEpubid mark.



% make the title area
\maketitle

%\newpage

\tableofcontents

\bigskip

%\begin{abstract}
%%\boldmath
%In this letter, an algorithm for evaluating the exact analytical bit error rate  (BER)  for the piecewise linear (PL) combiner for  multiple relays is presented. Previous results were available only for upto three relays. The algorithm is unique in the sense that  the actual mathematical expressions, that are prohibitively large, need not be explicitly obtained. The diversity gain due to multiple relays is shown through plots of the analytical BER, well supported by simulations. 
%
%\end{abstract}
\begin{abstract}
%\boldmath
This manual provides a beginner level application of signal processing by filtering noise from an audio signal recorded using a mobile phone.  A built-in Python module for the Butterworth low pass filter (LPF) is used for filtering out noise present in higher frequencies. Through this application, relevant concepts in DSP are explored.
\end{abstract}
% IEEEtran.cls defaults to using nonbold math in the Abstract.e 
% This preserves the distinction between vectors and scalars. However,
% if the journal you are submitting to favors bold math in the abstract,
% then you can use LaTeX's standard command \boldmath at the very start
% of the abstract to achieve this. Many IEEE journals frown on math
% in the abstract anyway.

% Note that keywords are not normally used for peerreview papers.
%\begin{IEEEkeywords}
%Cooperative diversity, decode and forward, piecewise linear
%\end{IEEEkeywords}



% For peer review papers, you can put extra information on the cover
% page as needed:
% \ifCLASSOPTIONpeerreview
% \begin{center} \bfseries EDICS Category: 3-BBND \end{center}
% \fi
%
% For peerreview papers, this IEEEtran command inserts a page break and
% creates the second title. It will be ignored for other modes.
\IEEEpeerreviewmaketitle


%\documentclass{article}
%\usepackage[utf8]{inputenc}
%\usepackage{listings}
%\usepackage{graphicx}

%\usepackage{color}
%\definecolor{codegreen}{rgb}{0,0.6,0}
%\definecolor{codegray}{rgb}{0.5,0.5,0.5}
%\definecolor{codepurple}{rgb}{0.58,0,0.82}
%\definecolor{backcolour}{rgb}{0.95,0.95,0.92}
%\lstdefinestyle{mystyle}{
    %backgroundcolor=\color{backcolour},   
    %commentstyle=\color{codegreen},
    %keywordstyle=\color{magenta},
    %numberstyle=\tiny\color{codegray},
    %stringstyle=\color{codepurple},
    %basicstyle=\footnotesize,
    %breakatwhitespace=false,         
    %breaklines=true,                 
    %captionpos=b,                    
    %keepspaces=true,                 
    %numbers=left,                    
    %numbersep=5pt,                  
    %showspaces=false,                
    %showstringspaces=false,
    %showtabs=false,                  
    %tabsize=2
%}
 
%\lstset{style=mystyle}


%\title{Analog Design Through Arduino}
%\author{G V V Sharma* }

%\begin{document}

%\maketitle
\section{Software Installation}
Run the following commands
\begin{lstlisting}
sudo apt-get update
sudo apt-get install libsndfile1
sudo apt-get install libffi-dev
sudo pip install pysoundfile
\end{lstlisting}
\section{Digital Filter}
\begin{problem}
\label{prob:input}
Download the sound file 
%\href{https://github.com/gadepall/EE1330/raw/master/intro/Sound_Noise.wav}{\url{https://github.com/gadepall/EE1330/raw/master/intro/Sound_Noise.wav}}  
\begin{lstlisting}
wget https://github.com/gadepall/EE1330/raw/master/intro/Sound_Noise.wav
\end{lstlisting}
%in the link given below.
%\linebreak
\end{problem}

\begin{problem}
\label{prob:spectrogram}
You will find a spectrogram at \href{https://academo.org/demos/spectrum-analyzer}{\url{https://academo.org/demos/spectrum-analyzer}}. 
%\end{problem}
%%
%
%%\onecolumn
%%\input{./figs/fir}
%\begin{problem}
Upload the sound file that you downloaded in Problem \ref{prob:input} in the spectrogram  and play.  Observe the spectrogram. What do you find?
\end{problem}
%
\solution There are a lot of yellow lines between 440 Hz to 5.1 KHz.  These represent the synthesizer key tones. Also, the key strokes
are audible along with background noise.
% By observing spectrogram, it clearly shows that tonal frequency is under 4kHz. And above 4kHz only noise is present.
\begin{problem}
\label{prob:output}
Write the python code for removal of out of band noise and execute the code.
\end{problem}
\solution
\lstinputlisting{Cancel_noise.py}
%\begin{figure}[h]
%\centering
%\includegraphics[width=\columnwidth]{enc_block_diag.png}
%\caption{}
%\label{fig:convolution encoder}
%\end{figure}
%\input{block_enc}
\begin{problem}
The output of the python script in Problem \ref{prob:output} is the audio file Sound\_With\_ReducedNoise.wav. Play the file in the spectrogram in Problem \ref{prob:spectrogram}. What do you observe?
\end{problem}
\solution The key strokes as well as background noise is subdued in the audio.  Also,  the signal is blank for frequencies above 5.1 kHz.
\linebreak
Answer the following questions by looking at the python code in Problem \ref{prob:output}.
\begin{problem}
What is the sampling frequency of the input signal?
\end{problem}
\solution
Sampling frequency(fs)=44.1kHZ.
\begin{problem}
What is type, order and  cutoff-frequency of the above butterworth filter
\end{problem}
\solution
The given butterworth filter is low pass with order=2 and cutoff-frequency=4kHz.
%
\begin{problem}
Modifying the code with different input parameters and to get the best possible output.
\end{problem}
%
\begin{problem}
The command
\begin{lstlisting}
	output_signal = signal.filtfilt(b, a, input_signal)
	\end{lstlisting}
in Problem \ref{prob:output} is executed through the following difference equation
\begin{equation}
\label{eq:iir_filter}
 \sum _{m=0}^{M}a\brak{m}y\brak{n-m}=\sum _{k=0}^{N}b\brak{k}x\brak{n-k}
\end{equation}
%
where the input signal is $x(n)$ and the output signal is $y(n)$ with initial values all 0. Replace
\textbf{signal.filtfilt} with your own routine and verify.
\end{problem}
%
\section{$Z$-transform}
\begin{problem}
The $Z$-transform of $x(n)$ is defined as
%
\begin{equation}
\label{eq:z_trans}
X(z)={\mathcal {Z}}\{x(n)\}=\sum _{n=-\infty }^{\infty }x(n)z^{-n}
\end{equation}
%
Show that
\begin{equation}
{\mathcal {Z}}\{x(n-1)\} = z^{-1}X(z)
\end{equation}
and find
\begin{equation}
\label{eq:z_trans_shift}
{\mathcal {Z}}\{x(n-k)\} 
\end{equation}
\end{problem}
%
\begin{problem}
Find
%
\begin{equation}
H(z) = \frac{Y(z)}{X(z)}
\end{equation}
%
from  \eqref{eq:iir_filter} assuming that the $Z$-transform is a linear operation.
\end{problem}
%
%
\begin{problem}
Find the Z transform of 
\begin{equation}
\delta(n)
=
\begin{cases}
1 & n = 0
\\
0 & \text{otherwise}
\end{cases}
\end{equation}
and show that the $Z$-transform of
\begin{equation}
\label{eq:unit_step}
u(n)
=
\begin{cases}
1 & n \ge 0
\\
0 & \text{otherwise}
\end{cases}
\end{equation}
is
\begin{equation}
U(z) = \frac{1}{1-z^{-1}}, \quad \abs{z} > 1
\end{equation}
\end{problem}

\begin{problem}
Show that 
\begin{equation}
\label{eq:anun}
a^nu(n) \ztrans \frac{1}{1-az^{-1}} \quad \abs{z} > \abs{a}
\end{equation}
\end{problem}
%
\begin{problem}
\label{prob:H(z)}
Obtain $H(z)$ for $b$ and $a$ in Problem \ref{prob:output} using \eqref{eq:z_trans_shift}. 
\end{problem}
\begin{problem}
Let
\begin{equation}
H\brak{e^{\j \omega}} = H\brak{z = e^{\j \omega}}.
\end{equation}
Plot $\abs{H\brak{e^{\j \omega}}}$ for $H(z)$ in Problem \ref{prob:H(z)}.  Comment.  $H(e^{\j \omega})$ is
known as the {\em Discret Time Fourier Transform} (DTFT) of $x(n)$.
\end{problem}
\begin{problem}
\label{prob:iir}
Show that $H(z)$ in Problem \ref{prob:H(z)} can be expressed as
%
\begin{equation}
\label{eq:ztransab}
H(z) = \sum_{k}\frac{c_k}{1-d_kz^{-1}}
\end{equation}
%
using partial fractions. Find the values of $c_k$ and $d_k$.
%the filter represented by the coefficients $a$ and $b$ in \eqref{eq:iir_filter} is defined as 
\end{problem}
\section{Impulse Response}
\begin{problem}
\label{prob:impulse_resp}
Find an expression for $h(n)$ using $H(z)$ in Problem \ref{eq:ztransab} and \eqref{eq:anun}, given that
\begin{equation}
\label{eq:impulse_resp}
h(n) \ztrans H(z)
\end{equation}
and there is a one to one relationship between $h(n)$ and $H(z)$. $h(n)$ is known as the {\em impulse response} of the
system defined by \eqref{eq:iir_filter}.
\end{problem}
%
\begin{problem}
Sketch $h(n)$. Is it bounded? Convergent? 
\end{problem}
\begin{problem}
The system with $h(n)$ is defined to be stable if
\begin{equation}
\sum_{n=-\infty}^{\infty}h(n) < \infty
\end{equation}
Is the system defined by \eqref{eq:iir_filter} stable for the impulse response in \eqref{eq:impulse_resp}?
\end{problem}
%
\begin{problem}
Compute $h(n)$ using 
\begin{equation}
\label{eq:iir_filter_h}
 \sum _{m=0}^{M}a\brak{m}h\brak{n-m}=\sum _{k=0}^{N}b\brak{k}\delta\brak{n-k}
\end{equation}
\end{problem}
This is the definition of $h(n)$.
\begin{problem}
Compute 
%
\begin{equation}
\label{eq:convolution}
y(n) = x(n)*h(n) = \sum_{n=-\infty}^{\infty}x(k)h(n-k)
\end{equation}
%
where $x(k)$ is the \textbf{input\_signal} in Problem \ref{prob:output}.  You will need to suitably truncate $h(n)$ calculated in Problem \ref{prob:impulse_resp}. Use $y(n)$ as \textbf{output\_signal} in Problem \ref{prob:output}.  Comment. The operation in \eqref{eq:convolution} is known as
{\em convolution}.
\end{problem}
%
\begin{problem}
Show that
\begin{equation}
y(n) =  \sum_{n=-\infty}^{\infty}x(n-k)h(k)
\end{equation}
\end{problem}
%
\section{DFT and FFT}
\begin{problem}
Compute
\begin{equation}
X(k) \define \sum _{n=0}^{N-1}x(n) e^{-\j2\pi kn/N}, \quad k = 0,1,\dots, N-1
\end{equation}
and $H(k)$ using $h(n)$.
\end{problem}
\begin{problem}
Compute 
\begin{equation}
Y(k) = X(k)H(k)
\end{equation}
\end{problem}
\begin{problem}
Compute
\begin{equation}
 y\brak{n}={\frac {1}{N}}\sum _{k=0}^{N-1}Y\brak{k}\cdot e^{\j 2\pi kn/N},\quad n = 0,1,\dots, N-1
\end{equation}
Use $y(n)$ as \textbf{output\_signal} in Problem \ref{prob:output}.
\end{problem}
\begin{problem}
Repeat the previous exercise by computing $X(k), H(k)$ and $y(n)$ through FFT and IFFT.
\end{problem}
%
\begin{problem}
Wherever possible, express all the above equations as matrix equations.
\end{problem}
%\begin{problem}
%Play the orignal sound file and filtered sound file?
%\end{problem}
%\solution
%\linebreak
%It can be obseved that out of band noise of original soundfile is eliminated.
\end{document}


