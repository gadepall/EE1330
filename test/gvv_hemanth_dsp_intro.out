\BOOKMARK [1][-]{section.1}{Software Installation}{}% 1
\BOOKMARK [1][-]{section.2}{Digital Filter}{}% 2
\BOOKMARK [1][-]{section.3}{Z-transform}{}% 3
\BOOKMARK [1][-]{section.4}{Impulse Response}{}% 4
\BOOKMARK [1][-]{section.5}{DFT and FFT}{}% 5
